sdk/storage/azure-storage-blobs/swagger/README.md

```diff
@@ -0,0 +1,1361 @@
+      $["x-ms-encryption-key-sha256"]["x-nullable"] = true;
+      $["x-ms-encryption-scope"]["x-nullable"] = true;
+      $["Content-MD5"]["x-ms-client-name"] = "TransactionalContentHash";
+      $["Content-MD5"]["x-nullable"] = true;
+      $["x-ms-content-crc64"]["x-ms-client-name"] = "TransactionalContentHash";
+      $["x-ms-content-crc64"]["x-nullable"] = true;
+```
+
+### CommitBlockList
+
+```yaml
+directive:
+  - from: swagger-document
+    where: $["x-ms-paths"]["/{containerName}/{blob}?comp=blocklist"].put.responses["201"].headers
+    transform: >
+      $["x-ms-version-id"]["x-nullable"] = true;
+      $["Content-MD5"]["x-ms-client-name"] = "TransactionalContentHash";
+      $["Content-MD5"]["x-nullable"] = true;
+      $["x-ms-content-crc64"]["x-ms-client-name"] = "TransactionalContentHash";
+      $["x-ms-content-crc64"]["x-nullable"] = true;
+      $["x-ms-encryption-key-sha256"]["x-nullable"] = true;
+      $["x-ms-encryption-scope"]["x-nullable"] = true;
+```
+
+### GetBlockList
+
+```yaml
+directive:
+  - from: swagger-document
+    where: $.definitions
+    transform: >
+      $.Block["x-ms-client-name"] = "BlobBlock";
+      $.Block["xml"] = {"name": "Block"};
+  - from: swagger-document
+    where: $["x-ms-paths"]["/{containerName}/{blob}?comp=blocklist"].get.responses
+    transform: >
+      $["200"].headers["x-ms-blob-content-length"]["x-ms-client-name"] = "BlobSize";
+      $["200"].headers["x-ms-blob-content-length"]["x-nullable"] = true;
+      $["200"].headers["ETag"]["x-ms-client-name"] = "ETag";
+      $["200"].headers["ETag"]["x-nullable"] = true;
+      $["200"].headers["Last-Modified"]["x-ms-client-name"] = "LastModified";
+      $["200"].headers["Last-Modified"]["x-nullable"] = true;
+      delete $["200"].headers["Content-Type"];
+      $["200"].schema = {
+        "x-ms-client-name": "GetBlockListResult",
+        "xml": {"name": "BlockList"},
+        "type": "object",
+        "properties": {
+          "ETag": {"type": "string", "format": "etag", "x-ms-xml": {"name": ""}, "description": "The ETag contains a value that you can use to perform operations conditionally. If the request version is 2011-08-18 or newer, the ETag value will be in quotes."},
+          "LastModified": {"type": "string", "format": "date-time-rfc1123", "x-ms-xml": {"name": ""}, "description": "Returns the date and time the container was last modified. Any operation that modifies the blob, including an update of the blob's metadata or properties, changes the last-modified time of the blob."},
+          "BlobSize": {"type": "integer", "format": "int64", "x-ms-client-default": "0", "x-ms-xml": {"name": ""}, "description": "Size of the blob in bytes."},
+          "CommittedBlocks": {"type": "array", "items": {"$ref": "#/definitions/Block"}, "description": "List of committed blocks."},
+          "UncommittedBlocks": {"type": "array", "items": {"$ref": "#/definitions/Block"}, "description": "List of uncommitted blocks."}
```

Comment:
what is this for?

Reply:
I am assuming your question is about the whole file, in general.

This is a readme transform file that modifies the swagger before being passed-in to the modeler 4 component of autorest (which is then passed into the language plugin that generates C++). This way, the C++ language plugin/extension can remain agnostic of "tweaks" or corrections we need to make to service specific swagger.

Reply:
If your question is specific to these particular lines your comment highlighted that are making the modifications, @Jinming-Hu can talk more to that.

Reply:
These lines are used to change the return type of function `BlockBlobClient::GetBlockList`. In the original swagger file, as you can see [here](https://github.com/Jinming-Hu/azure-storage-api-specs/blob/main/Microsoft.BlobStorage/preview/2020-02-10/blob.json#L7076) and [here](https://github.com/Jinming-Hu/azure-storage-api-specs/blob/73a91e111c677d392d8459f6e06ac1b7f2ff4a3e/Microsoft.BlobStorage/preview/2020-02-10/blob.json#L11240), the return type is

```C++
struct XXX {
  std::vector<BlockList>CommittedBlocks;
  std::vector<BlockList> UncommittedBlocks;
};
```

but in the existing code, the return type is

https://github.com/Azure/azure-sdk-for-cpp/blob/main/sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/protocol/blob_rest_client.hpp#L1918